ffi/SDL2_0.lua

```diff
@@ -435,4 +435,39 @@ function S.gameControllerRumble(left_intensity, right_intensity, duration)
     return SDL.SDL_GameControllerRumble(S.controller, left_intensity, right_intensity, duration)
 end
 
+function S.getPlatform()
+    return ffi.string(SDL.SDL_GetPlatform())
+end
+
+function S.getBasePath()
+    return ffi.string(SDL.SDL_GetBasePath())
+end
+
+function S.getPrefPath(organization, appname)
+    if not organization then organization = "dummy" end
+    if not appname then appname = "application" end
+    return ffi.string(SDL.SDL_GetPrefPath(organization, appname))
+end
+
+function S.getPowerInfo()
+    local batt, plugged, charging
+    local ptr = ffi.new("int[1]", {0})
+    local battery_info = SDL.SDL_GetPowerInfo(nil, ptr)
+    if battery_info == SDL.SDL_POWERSTATE_UNKNOWN
+    or battery_info == SDL.SDL_POWERSTATE_NO_BATTERY then
```

Comment:
Btw, we had a discussion about this last week, see https://github.com/koreader/koreader/pull/6364#issuecomment-658361332 and https://github.com/koreader/koreader/pull/6364#issuecomment-658367766

@yparitcher went with
```lua
                        if caller[location][settings] ~= nil
                            and caller[location][settings][k]
                        then
```